clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesPodMetric.java

```diff
@@ -16,26 +16,51 @@
 
 package com.netflix.spinnaker.clouddriver.kubernetes.v2.description;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import java.util.ArrayList;
-import java.util.List;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
 import java.util.Map;
+import java.util.Optional;
+import javax.annotation.ParametersAreNullableByDefault;
 import lombok.*;
 
-@Getter
-@Builder
+@NonnullByDefault
+@Value
 public class KubernetesPodMetric {
   private final String podName;
   private final String namespace;
-  @Builder.Default private final List<ContainerMetric> containerMetrics = new ArrayList<>();
+  private final ImmutableList<ContainerMetric> containerMetrics;
 
-  @Data
   @Builder
-  @NoArgsConstructor
-  @AllArgsConstructor
+  @ParametersAreNullableByDefault
+  public KubernetesPodMetric(
+      String podName, String namespace, Iterable<ContainerMetric> containerMetrics) {
+    this.podName = Strings.nullToEmpty(podName);
```

Comment:
I had never seen `nullToEmpty` before; how great! ⛔️ ➡️ 🔤

Reply:
Yeah, I just learned about it too!  There are a lot of useful things hiding in guava!